InertiaCore/Utils/Vite.cs

```diff
@@ -0,0 +1,147 @@
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+namespace InertiaCore.Utils;
+
+public static class Vite
+{
+
+    private static string GetString(IHtmlContent content)
+    {
+        var writer = new System.IO.StringWriter();
+        content.WriteTo(writer, HtmlEncoder.Default);
+        return writer.ToString();
+    }
+
+    private static string publicDir(string path)
+    {
+        return "wwwroot/build/" + path;
+    }
+
+    public static HtmlString input(string path)
+    {
+        if (isRunningHot())
+        {
+            return new HtmlString(makeModuleTag(hotAsset("@vite/client")) + makeModuleTag(hotAsset(path)));
+        }
+
+        if (!File.Exists(publicDir("manifest.json")))
+        {
+            throw new Exception("Vite Manifest is missing. Run `npm run build` and try again.");
+        }
+
+        var manifest = File.ReadAllText(publicDir("manifest.json"));
+        var manifestJson = JsonSerializer.Deserialize<Dictionary<string, JsonElement>>(manifest);
+
+        if (manifestJson == null)
+        {
+            throw new Exception("Vite Manifest is invalid. Run `npm run build` and try again.");
+        }
+
+        if (!manifestJson.ContainsKey(path))
+        {
+            throw new Exception("Asset not found in manifest: " + path);
+        }
+
+        JsonElement obj = manifestJson[path];
+        var filePath = obj.GetProperty("file");
+
+        if (filePath.ToString().EndsWith(".css"))
```

Comment:
There are much more possible `css` file extensions. Below is the method from `Vite.php` which checks it using regular expression:
```php
protected function isCssPath($path)
{
    return preg_match('/\.(css|less|sass|scss|styl|stylus|pcss|postcss)$/', $path) === 1;
}
```